CONTRIBUTING.md

```diff
@@ -3,62 +3,75 @@
+### Commands & Toolchain
 
-[ESLint](https://eslint.org/) is used for JavaScript files. The rules are primarily inherited from the Airbnb rules and made to work with the Prettier settings, with some minor project-specific modifications. If possible, it can be helpful to have your editor lint JavaScript files on the fly.
+The project includes a `Makefile` to centralize various commands used in development and operation. When new commands are required, be sure to add them to the `Makefile`, even if it calls out to something else, like `npm run-script`.
 
-CloudFormation templates should be run through [cfn-python-lint](https://github.com/aws-cloudformation/cfn-python-lint) and be free of linting errors. Your editor should check your files in realtime if possible; there are plugins available for many popular editors.
+Besides language runtimes, package managers, and command line programs, tooling should be written to assume that libraries are installed within the package, not globally available on the system. I.e, do not asusme the `eslint` or `prettier` are installed globally.
 
-### AWS
+## Getting Started
 
-There are a number of shell scripts in this repo. They assume you have a profile in `~/.aws/config` called `prx-legacy`, which is configured to access the `prx-legacy` AWS account.
+After cloning the Git repository for Porter, run `make bootstrap` to install all necessary development dependencies. This requires already having Node.js, Ruby, and Python 3 installed on your system.
 
 ## Deploying
 
-The template is intended to deployed using the [SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-command-reference.html). You can run `sam deploy` if you already have a `samconfig.toml`, or `sam deploy -g` if you don't. Additionally, there is a VS Code task that supports deploying various environments (staging, production, etc), which expects files called `samconfig.[stag|prod].toml`, and will use the correct one.
+See the [INSTALL](https://github.com/PRX/Porter/blob/master/INSTALL.md) guide for detailed information about deploying Porter to AWS.
 
-## Misc
+## State Machine Guidelines
 
 Various APIs and SDKs are inconsistent in how they refer to S3 buckets and objects. To be consistent throughout this project, they should be called `BucketName` and `ObjectKey` until they are passed to an API.
 
-For the sake of clarity, when building states always include the `InputPath`, `ResultPath`, and `OutputPath`, even if they are set to the default value.
+For the sake of clarity, when building states in ASL always include the `InputPath`, `ResultPath`, and `OutputPath`, even if they are set to the default value.
 
 Timestamps should always be in seconds.
 
-Create Lambda Layers for external dependencies, rather than including them in the deployment package.
+Create Lambda Layers for external dependencies. Do not include packages and libraries in the deployable Lambda code iteself.
```

Comment:
Is this a standard practice now?  For a small-ish list of dependencies, I think you still get better cold start times by bundling with the code.  And you save a bit of complexity.

(I'm also annoyed that edge lambdas don't support layers yet).

Is it also worth noting you should only include the aws-sdk as a dev dependency?  Since it's already available in the lambda execution for free.

Reply:
Initially, before this projected migrated to AWS SAM deploys, the script I was using for deploys was not smart enough to build external dependencies into the deployment package, so it was ~required to use layers. With SAM it is now much simpler to include packages in the Zip file, so potentially we could start doing that. 

Most of the dependencies that exist are things like `file-type`, `ffmpeg`, etc, which more than one state use, and I find it quite a cleaner to manage those as layers to ensure that all the states are using the same version of those libraries.

I could imagine someday managing dependencies both ways, but for now I think layers are fine for everything, given how infrequently things are changing.